### ISSUE TITLE
IllegalStateException("A task may only be set once.")

### ISSUE THREAD

**Issue by kakinIA** [airbnb/lottie-android#937]
i used the lottie animation in recyclerview. However,there was a crash "A task may only be set once" happened in sometime.I wanted to know the situation about it. When is the LottieTask#result not null but execute "startTaskObserverIfNeeded()" ? 

here is the layout
```
        <com.airbnb.lottie.LottieAnimationView
            android:id="@+id/lav"
            android:layout_width="@dimen/dimens_18dp"
            android:layout_height="@dimen/dimens_18dp"
            android:layout_gravity="end|center_vertical"
            android:scaleType="centerInside"
            app:lottie_cacheStrategy="weak"
            app:lottie_rawRes="@raw/anim_follow" />
```
and i just use custom animator when i click the view, like this
```
// Custom animation speed or duration.
ValueAnimator animator = ValueAnimator.ofFloat(0f, 1f);
animator.addUpdateListener(animation -> {
    animationView.setProgress(animation.getAnimatedValue());
});
animator.start();

```
at the same time, set the progress according to the status in onBindViewHolder()

**Comment by gpeal** [comment]
@kakinIA Please attach the full stacktrace

**Comment by kakinIA** [comment]
@gpeal 

```
 java.lang.IllegalStateException
A task may only be set once.
cy.a(LottieTask.java:72)
1 cy.a(LottieTask.java:72)
2 cy.a(LottieTask.java:28)
3 cy$2.run(LottieTask.java:194)
```

**Comment by gpeal** [comment]
@kakinIA That's not helpful. Can you provide one with an unobfuscated stack trace?

**Comment by kakinIA** [comment]
i'm sorry, because the version is release and catch the stack trace in tencent bugly which is statistics exception platform.
```
java.lang.IllegalStateException

A task may only be set once.
1 com.airbnb.lottie.LottieTask.void setResult(com.airbnb.lottie.LottieResult)(LottieTask.java:72)
--
2 com.airbnb.lottie.LottieTask.com.airbnb.lottie.LottieResult access$000(com.airbnb.lottie.LottieTask)(LottieTask.java:28)
3 ##_parent_##3##_parent_##
4 ##_child_## void access$200(com.airbnb.lottie.LottieTask,java.lang.Object)##_child_##
5 ##_child_## void access$300(com.airbnb.lottie.LottieTask,java.lang.Throwable)##_child_##
6 ##_child_## void access$400(com.airbnb.lottie.LottieTask,com.airbnb.lottie.LottieResult)##_child_##
7 com.airbnb.lottie.LottieTask$2.void run()(LottieTask.java:194)
```

**Comment by gpeal** [comment]
@kakinIA I've  added some additional protection against this in https://github.com/airbnb/lottie-android/pull/933

I'll release 2.6.1 today

**Comment by gpeal** [comment]
Fixed by #933
